src/gic_v2.rs

```diff
@@ -111,70 +125,234 @@ pub struct GicDistributor {
 /// - setting an interrupt priority mask for the processor
 /// - defining the preemption policy for the processor
 /// - determining the highest priority pending interrupt for the processor.
+
+/* 
+#[derive(Debug, Clone)]
 pub struct GicCpuInterface {
     base: NonNull<GicCpuInterfaceRegs>,
 }
+*/
+
+// type GicCpuInterface = GicCpuInterfaceRegs;
+
 
 unsafe impl Send for GicDistributor {}
 unsafe impl Sync for GicDistributor {}
 
 unsafe impl Send for GicCpuInterface {}
 unsafe impl Sync for GicCpuInterface {}
 
+use crate::device_ref::*;
+
+static mut GICD: DeviceRef<GicDistributor> = unsafe { DeviceRef::new() };
```

Comment:
Instantiate it in axhal of arceos, just as we discussed in the meeting yesterday.